content/en/docs/guides/cron-job-image-auth.md

```diff
@@ -0,0 +1,377 @@
+            imagePullPolicy: IfNotPresent
+            # You will need to set the AWS_ACCESS_KEY_ID and AWS_SECRET_ACCESS_KEY environment variables if not using
+            # IRSA. It is recommended to store the values in a Secret and load them in the container using envFrom.
+            # envFrom:
+            # - secretRef:
+            #     name: aws-credentials
+            env:
+            - name: REGION
+              value: us-east-1 # change this if ECR repo is in a different region
+            volumeMounts:
+            - mountPath: /token
+              name: token
+            command:
+            - /bin/sh
+            - -ce
+            - aws ecr get-login-password --region ${REGION} > /token/ecr-token
+          containers:
+          - image: bitnami/kubectl
+            name: create-secret
+            imagePullPolicy: IfNotPresent
+            env:
+            - name: SECRET_NAME
+              value: ecr-credentials
+            - name: ECR_REGISTRY
+              value: <account id>.dkr.ecr.<region>.amazonaws.com # fill in the account id and region
+            volumeMounts:
+            - mountPath: /token
+              name: token
+            command:
+            - /bin/bash
+            - -ce
+            - |-
+              kubectl create secret docker-registry $SECRET_NAME \
+                --dry-run=client \
+                --docker-server="$ECR_REGISTRY" \
+                --docker-username=AWS \
+                --docker-password="$(</token/ecr-token)" \
+                -o yaml | kubectl apply -f -
+```
+
+{{% alert color="info" title="Using IAM Roles for Service Accounts (IRSA)" %}}
+If using IRSA, make sure the role attached to the service account has
+readonly access to ECR. The AWS managed policy
+`arn:aws:iam::aws:policy/AmazonEC2ContainerRegistryReadOnly` can be attached
+to the role.
+{{% /alert %}}
+
+Since the cronjob will not create a job right away, after applying the manifest,
+you can manually create an init job using the following command:
+
+```sh
+$ kubectl create job --from=cronjob/ecr-credentials-sync -n flux-system ecr-credentials-sync-init
+```
+
+After the job runs, a secret named `ecr-credentials` should be created. Use this
+name in your ECR ImageRepository resource manifest as the value for
+`.spec.secretRef.name`.
+
+```yaml
+spec:
+  secretRef:
+    name: ecr-credentials
+```
+
+### GCP Container Registry
+
+#### Using Native GCP GCR Auto-Login
+
+#### Using access token [short-lived]
+
+{{% alert color="info" title="Workload Identity" %}}
+Please ensure that you enable workload identity for your cluster, create a GCP service account that has
+access to the container registry and create an IAM policy binding between the GCP service account and
+the Kubernetes service account so that the pods created by the cronjob can access GCP APIs and get the token.
+Take a look at [this guide](https://cloud.google.com/kubernetes-engine/docs/how-to/workload-identity)
```

Comment:
As a newbie to GCP Workload Identity, I was able to follow the abridged GCR instructions solely because of this link, it was too complicated without all four main steps being laid out explicitly as done here. I also could have used a primer on GCP IAM roles, as I'm pretty sure it's possible to get this part wrong, but the wizard seems to have created the right thing for me based on selecting "Container Registry Service Agent".

Maybe we should not be in the business of onboarding GKE customers to their IDP, but after this feature merges I'd like to take a stab at a guide that will cover this information in more detail again.

In the mean time, we should make sure the "[this guide](https://cloud.google.com/kubernetes-engine/docs/how-to/workload-identity)" link is included in the Workload Identity / Autologin instructions below as well. IMHO 👍 this is a good structure for this change, and it makes it clear we understand that autologin is likely to be the preferred and ideal mode for most users.

Even if it is not ideal for (soft) multi-tenancy, I think that is the reality on the ground.

Reply:
Resolved with 1f9240b